src/overlays/actors/ovl_En_Dog/z_en_dog.c

```diff
@@ -22,41 +28,579 @@ const ActorInit En_Dog_InitVars =
+{
+    u32* walk = &D_06001368;
+    if (this->skelAnime.animCurrent == walk)
+    {
+        if ((this->skelAnime.animCurrentFrame == 1.0f) || (this->skelAnime.animCurrentFrame == 7.0f))
+        {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHIBI_WALK);
+        }
+    }
+}
+
+static void EnDog_RunSFX(EnDog* this)
+{
+    u32* run = &D_06000D78;
+    if (this->skelAnime.animCurrent == run)
+    {
+        if ((this->skelAnime.animCurrentFrame == 2.0f) || (this->skelAnime.animCurrentFrame == 4.0f))
+        {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHIBI_WALK);
+        }
+    }
+}
+
+static void EnDog_BarkSFX(EnDog* this)
+{
+    u32* bark = &D_06000278;
+    if (this->skelAnime.animCurrent == bark)
+    {
+        if ((this->skelAnime.animCurrentFrame == 13.0f) || (this->skelAnime.animCurrentFrame == 19.0f))
+        {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_SMALL_DOG_BARK);
+        }
+    }
+}
+
+static s32 EnDog_PlayAnimSFX(EnDog* this)
+{
+    s32 animation;
+
+    if (this->behavior != this->nextBehavior)
+    {
+        if (this->nextBehavior == DOG_SIT_2)
+        {
+            this->nextBehavior = DOG_SIT;
+        }
+        if (this->nextBehavior == DOG_BOW_2)
+        {
+            this->nextBehavior = DOG_BOW;
+        }
+
+        this->behavior = this->nextBehavior;
+        switch(this->behavior)
+        {
+            case DOG_WALK: 
+                animation = 1; 
+                break;
+            case DOG_RUN: 
+                animation = 2; 
+                break;
+            case DOG_BARK: 
+                animation = 3; 
+                break;
+            case DOG_SIT:
+                animation = 4; 
+                break;
+            case DOG_BOW: 
+                animation = 6; 
+                break;
+        }
+        func_80034EC0(&this->skelAnime, &animations, animation);
+    }
+    
+    switch(this->behavior)
+    {
+        case DOG_SIT: 
+            if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount) != 0)
+            {
+                func_80034EC0(&this->skelAnime, &animations, 5);
+                this->behavior = this->nextBehavior = DOG_SIT_2;
+            }
+            break;
+        case DOG_BOW: 
+            if (func_800A56C8(&this->skelAnime, this->skelAnime.animFrameCount) != 0)
+            {
+                func_80034EC0(&this->skelAnime, &animations, 7);
+                this->behavior = this->nextBehavior = DOG_BOW_2;
+            }
+            break;
+        case DOG_WALK: 
+            EnDog_WalkSFX(this); 
+            break;
+        case DOG_RUN: 
+            EnDog_RunSFX(this);
+            break;
+        case DOG_BARK: 
+            EnDog_BarkSFX(this);
+            if (this){} // needed for regalloc
+            break;
+    }
+    return 0;
+}
+
+static s8 EnDog_CanFollow(EnDog* this, GlobalContext* globalCtx)
+{
+    if ((this->collider.base.collideFlags & 2) != 0)
```

Comment:
This suggestion applies to many other places but != 0 and == 0 aren't ideal in boolean checks.
This is something we should try to get more consistent at. Sometimes there is no way to know for sure that it's a boolean check, but when it's obvious we should try to do it consistently.
```suggestion
    if (this->collider.base.collideFlags & 2)
```